src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -5,56 +5,56 @@
+    /// <summary>
+    /// HttpClient based implementation of DataServiceClientRequestMessage.
+    /// </summary>
+    public class HttpWebRequestMessage : DataServiceClientRequestMessage, IDisposable
     {
-        #region Private Fields
-
-#if PORTABLELIB
-        /// <summary> Running on silverlight </summary>
-        internal static bool IsRunningOnSilverlight = false;
-
-        /// <summary>Cached method info that will be used to call to set the userAgent property if its not null. </summary>
-        private static MethodInfo UserAgentMethodSetter;
-
-        /// <summary>Cached method info that will be used to call to set the ContentLength property if its not null. </summary>
-        private static MethodInfo ContentLengthMethodSetter;
-
-        /// <summary> Indicates the current platform the portablelib is running on can't set the user agent property </summary>
-        private static bool SettingUserAgentPropertyValidOnCurrentPlatform = true;
-
-        /// <summary> Indicates that the current platform doesn't support setting the header useragent property </summary>
-        private static bool SettingUserAgentOnHeadersCollectionValidOnCurrentPlatform = true;
-
-        /// <summary> Indicates the current platform the portablelib is running on can't set the user agent property </summary>
-        private static bool SettingContentLengthPropertyValidOnCurrentPlatform = true;
+        /// <summary>
+        /// HttpClient distinguishes "Content" headers from "Request" headers, so we
+        /// need to know which category a header belongs to.
+        /// </summary>
+        private static readonly HashSet<string> _contentHeaderNames = new HashSet<string> (new[] {
+            "Allow",
+            "Content-Disposition",
+            "Content-Encoding",
+            "Content-Language",
+            "Content-Length",
+            "Content-Location",
+            "Content-MD5",
+            "Content-Range",
+            "Content-Type",
+            "Expires",
+            "Last-Modified",
+        }, StringComparer.OrdinalIgnoreCase);
 
-        /// <summary> Indicates that the current platform doesn't support setting the header useragent property </summary>
-        private static bool SettingContentLengthOnHeadersCollectionValidOnCurrentPlatform = true;
-#endif
         /// <summary>Request Url.</summary>
-        private readonly Uri requestUrl;
+        private readonly Uri _requestUrl;
 
         /// <summary> The effective HTTP method. </summary>
-        private readonly string effectiveHttpMethod;
+        private readonly string _effectiveHttpMethod;
+
+        /// <summary>HttpRequestMessage instance.</summary>
+        private readonly HttpRequestMessage _requestMessage;
+        private readonly HttpClient _client;
+        private readonly HttpClientHandler _handler;
+        private readonly MemoryStream _messageStream;
+        private readonly Dictionary<string, string> _contentHeaderValueCache;

```

Comment:
@ElizabethOkerio May you could add a docstring on some of these fields where the usage is not obvious, e.g. `_contentHeaderValueCache`